analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd

```diff
@@ -0,0 +1,333 @@
+  spread(OS_status, n,
+         # any missing values (indicating 0 deceased or 0 living) should be 0, not NA
+         fill = 0) %>%
+  filter(DECEASED >= 3) 
+# show result: there are 7 groups to keep
+cancer_groups_of_interest_df
+
+# Subset to only the cancer groups with potentially suitable sample size
+meta_indep_cancer_groups <- meta_indep %>%
+  filter(cancer_group %in% cancer_groups_of_interest_df$cancer_group) %>%
+  # set reference level to Medulloblastoma
+  mutate(cancer_group = fct_relevel(cancer_group, "Medulloblastoma"))
+
+# How many patients got removed?
+nrow(meta_indep) - nrow(meta_indep_cancer_groups)
+```
+
+## Univariate Models
+
+ 
+Here, we run a survival analyses for several _independent_ predictors, one at a time. We use cox regression for all analyses (including categorical), as demonstrated in `survival-analysis_template.Rmd`.
+
++ `tp53_score` (continuous)
++ `telomerase_score` (continuous)
++ `hgg_group` (categorical)
++ `cancer_group` (categorical)
+
+
+
+### TP53 
+
+
+```{r cox_tp53}
+fit_save_model(meta_indep, 
+              "tp53_score", 
+             file.path(results_dir, "cox_terms_tp53.RDS"))
+
+```
+
+
+### Telomerase score
+
+```{r cox_telo}
+fit_save_model(meta_indep, 
+              "telomerase_score", 
+             file.path(results_dir, "cox_terms_telomerase.RDS"))
+```
+
+
+###  HGG group
+
+> non-HGAT is the reference level.
+
+```{r cox_hgg}
+fit_save_model(meta_indep_hgg, 
+              "hgg_group", 
+             file.path(results_dir, "cox_terms_hgg_group.RDS"))
+```
+
+### Cancer group (cox)
+
+> MB is the reference level
+
+```{r cox_cancer_group}
+fit_save_model(meta_indep_cancer_groups, 
+               "cancer_group", 
+               file.path(results_dir, "cox_terms_cancer_group.RDS"))
+```
+
+
+
+### Cancer group (log rank)
+
+Here, we perform a log-rank analysis specifically for the `cancer_group` predictor, using data that only considers the 7 cancer groups determined to have suitable sample sizes (`DECEASED >= 3`). The output below shows:
+
++ Table of model coefficients
++ Matrix of `Pairwise comparisons using Log-Rank test`, which shows which cancer groups are significantly different from one another
+
+Note that confidence intervals are so large that survival plots are not clearly legible, so we don't make that visualization here.
+
+```{r cancer_group_log_rank}
+
+# Print model coefficients
+fit_save_model(meta_indep_cancer_groups,
+               "cancer_group",
+               file.path(results_dir,
+                         "log_rank_survival_per_cancer_group.RDS"),
+               test = "log.rank")
+
+
+# Compare pairwise with Bonferroni correction for multiple testing
+survminer::pairwise_survdiff(survival::Surv(time = OS_years, event = OS_status) ~ cancer_group,
+                             meta_indep_cancer_groups %>% mutate(OS_status = ifelse(OS_status == "LIVING", 0, 1)), 
+                             p.adjust.method = "bonferroni")
```

Comment:
The `survival_analysis()` function was originally created to do the recoding within the function and expects "LIVING" and "DECEASED" - 
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/51e4fca33eaa306ec7e63291217dc03551bb1e13/analyses/survival-analysis/util/survival_models.R#L97 

I suppose the code would still work if this were already recoded, but I thought it would be preferable to follow with the precedent that was set in the [template Rmd](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/51e4fca33eaa306ec7e63291217dc03551bb1e13/analyses/survival-analysis/survival-analysis_template.Rmd) not to recode.